blocks/library/html/index.js

```diff
@@ -39,6 +42,9 @@ registerBlockType( 'core/html', {
 			this.state = {
 				preview: false,
 			};
+			const allowedHtmlTags = new Set( Object.keys( wp.editor.allowedPostHtml ) );
```

Comment:
- Can we be formatting this data from the server to be in a more useful shape? i.e. would we always be expecting to call `Object.keys` and if so, pass from the server as `array_keys` ? Not sure how likely it is we'd do anything with the allowed attributes and could avoid increasing the size of the page by including only what's used.
- The cache assignment here reminds me of [a since-removed document about props in state](http://web.archive.org/web/20160430110308/https://facebook.github.io/react/tips/props-in-getInitialState-as-anti-pattern.html), which is more to the point of: rather than splitting the source of truth, merely generate this rendering on the fly. Since `allowedPostHtml` is not a prop and is not likely to change, this may not be an issue, but could still be seen as an over-optimization to split the logic as we've done here.

Reply:
> Can we be formatting this data from the server to be in a more useful shape?

Actually it is useful as it is. The data consists of keys for the HTML tags, and then the values are objects that indicate the attributes that are allowed for each element. It's the same structure as: https://github.com/WordPress/wordpress-develop/blob/1eda9654da06715798cbf911372c0c13e5baf7ca/src/wp-includes/kses.php#L53-L416